go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -33,11 +33,8 @@ type nodePortWatcher struct {
 	nodeIP      *net.IPNet
 }
 
+// AddService handles configuring shared gateway bridge flows to steer External IP, Node Port, Ingress LB traffic into OVN
 func (npw *nodePortWatcher) AddService(service *kapi.Service) {
-	if !util.ServiceTypeHasNodePort(service) && len(service.Spec.ExternalIPs) == 0 {
```

Comment:
why removing this?
This was filtering all services that are not NodePort or LoadBalancer and does't have ExternalIPs, I think that are the only ones that nodes should care about

Reply:
what about a cluster ip service with no external ip and does not have node port, but does have load balancer ingress?

Reply:
hmm, should that be allowed?

Reply:
great catch Tim, you' ve found a bug in Kubernetes

Reply:
https://github.com/kubernetes/kubernetes/issues/96724

Reply:
but no, no others than loadbalancer type should be allowed to set the status

Reply:
and it won' t https://github.com/kubernetes/kubernetes/pull/96726

Reply:
however, the conclusion is that it is possible, to not break current API :shrug: so
```
if !util.ServiceTypeHasNodePort(service) && ( len(service.Spec.ExternalIPs) == 0 || len(service.Status.LoadBalancer.Ingress) == 0)
```

Reply:
```
	// don't process headless service or services that doesn't have NodePorts or ExternalIPs
	if !util.ServiceTypeHasClusterIP(svc) || !util.IsClusterIPSet(svc) {
		return nil
	}
```

Reply:
ok will add

Reply:
Do we want to add a note on how to properly deal with this when the API gets updated sufficently?